### PR TITLE
rename ShutdownController to ShutdownWatcher

### PR DESCRIPTION
**Pull request by jsvd**
slightly adjust the metaphor for the class that is responsible for
monitoring a pipeline's shutdown process, report on its status and
intervene if it's stalled

### CODE REVIEW

File: logstash-core/spec/logstash/shutdown_watcher_spec.rb [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-require "logstash/shutdown_controller"
+require "logstash/shutdown_watcher"
 
-describe LogStash::ShutdownController do
+describe LogStash::ShutdownWatcher do
 
   let(:check_every) { 0.01 }
   let(:check_threshold) { 100 }
-  subject { LogStash::ShutdownController.new(pipeline, check_every) }
+  subject { LogStash::ShutdownWatcher.new(pipeline, check_every) }
```
Comment (ph) [inline logstash-core/spec/logstash/shutdown_watcher_spec.rb]: you can use `described_class`  here so the only change would have been the describe block

**Comment by suyograo** [comment]
@jsvd ping. would like to close this out for 2.2

File: logstash-core/spec/logstash/shutdown_watcher_spec.rb (hunk 1 above)
Comment (jsvd) [inline logstash-core/spec/logstash/shutdown_watcher_spec.rb]: I know, but I really don't like "described_class", it's a terrible name, while having the class explicitly there makes it more readable IMO.
That said, I can change you feel strongly on this.

Comment (ph) [inline logstash-core/spec/logstash/shutdown_watcher_spec.rb]: I dont feel strong about it, just make it easier to refactor names :D

**Comment by ph** [comment]
LGTM, I prefer `Watcher` instead of the controller, I think it describe more that the class is actually doing.

**Comment by jsvd** [comment]
Closed by
master (be91439)
2.x (836482e)
2.2 (b887705)

**Pull request closed without merging**